src/nnet3/nnet-compile-utils.cc

```diff
@@ -598,5 +598,44 @@ bool HasContiguousProperty(
   return true;
 }
 
+
+// see comment in header.
+void GetNxList(const std::vector<Index> &indexes,
+               std::vector<std::pair<int32, int32> > *pairs) {
+  // set of (n,x) pairs
+  std::unordered_set<std::pair<int32, int32>, PairHasher<int32> > n_x_set;
+
+  for (std::vector<Index>::const_iterator iter = indexes.begin();
+       iter != indexes.end(); ++iter)
+    n_x_set.insert(std::pair<int32, int32>(iter->n, iter->x));
+  pairs->clear();
+  pairs->reserve(n_x_set.size());
+  for (std::unordered_set<std::pair<int32, int32>, PairHasher<int32> >::iterator
+           iter = n_x_set.begin(); iter != n_x_set.end(); ++iter)
+    pairs->push_back(*iter);
+  std::sort(pairs->begin(), pairs->end());
+}
+
+
+// see comment in header.
+void GetTList(const std::vector<Index> &indexes,
+              std::vector<int32> *t_values) {
+  // set of t values
+  std::unordered_set<int32> t_set;
```

Comment:
Optionally, use std::set, which is sorted. Might be marginally more efficient.

You may also use some stl magic to reduce the function to just three lines, arguably more readable:
```c++
std::set<int32> t_set;
std::remove_copy(indexes.begin(), indexes.end(), std::inserter(t_set, t_set.begin()), kNoTime);
t_values->assign(t_set.begin(), t_set.end());
```
(include `<algorithm>` and `<iterator>`).